### ISSUE TITLE
reCaptcha v2 & "Could not verify you are not a robot" msg

### ISSUE THREAD

**Issue by jagbarcelo** [unosquare/passcore#487]
**PassCore Server 4.2.2**
- OS: Windows
- Provider: Active Directory
- Settings file: the original default options with SiteKey and PrivateKey entries for reCaptcha
- Log file:

```
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/1.1 POST https://mypassword.reset.site/api/password application/json 700
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
      Executing endpoint 'Unosquare.PassCore.Web.Controllers.PasswordController.Post (Unosquare.PassCore.Web)'
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[3]
      Route matched with {action = "Post", controller = "Password"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] Post(Unosquare.PassCore.Web.Models.ChangePasswordModel) on controller Unosquare.PassCore.Web.Controllers.PasswordController (Unosquare.PassCore.Web).
warn: Unosquare.PassCore.Web.Controllers.PasswordController[0]
      Invalid Recaptcha
System.Text.Json.JsonException: The JSON value could not be converted to System.Collections.Generic.Dictionary`2[System.String,System.String]. Path: $.success | LineNumber: 1 | BytePositionInLine: 17.
 ---> System.InvalidOperationException: Cannot get the value of a token type 'True' as a string.
   at System.Text.Json.Utf8JsonReader.GetString()
   at System.Text.Json.Serialization.Converters.JsonConverterString.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnReadEnumerable(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.ReadEnumerable(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.HandleValue(JsonTokenType tokenType, JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& readStack, Utf8JsonReader& reader, Exception ex)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ParseCore(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Unosquare.PassCore.Web.Controllers.PasswordController.ValidateRecaptcha(String recaptchaResponse) in C:\Unosquare\passcore\src\Unosquare.PassCore.Web\Controllers\PasswordController.cs:line 153
   at Unosquare.PassCore.Web.Controllers.PasswordController.Post(ChangePasswordModel model) in C:\Unosquare\passcore\src\Unosquare.PassCore.Web\Controllers\PasswordController.cs:line 100
info: Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor[1]
      Executing ObjectResult, writing value of type 'Unosquare.PassCore.Web.Models.ApiResult'.
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[2]
      Executed action Unosquare.PassCore.Web.Controllers.PasswordController.Post (Unosquare.PassCore.Web) in 392.16310000000004ms
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint 'Unosquare.PassCore.Web.Controllers.PasswordController.Post (Unosquare.PassCore.Web)' 
```

**Describe the bug**
Whenever I enable reCaptcha (v2) and the challenge 'I'm not a robot' box appears, PassCore is not able to change user's password and fails with the message 'Could not verify you are not a robot'. In reality, this error message shows to the user in the browser hides the real problem which is the exception thrown during ValidateRecaptcha(String recaptchaResponse) method.

Without reCaptcha, PassCore works fine and users are able to update their passwords.

According to reCaptcha documentation (see https://developers.google.com/recaptcha/docs/verify ) the call to https://www.google.com/recaptcha/api/siteverify is to be done with a POST. However, in the file PasswordController.cs, (method ValidateRecaptcha) the call is done with a GET:

`var response = await client.GetStringAsync(requestUrl);`

**Comment by jagbarcelo** [comment]
This piece of code fixes the issue:

```
using var client = new HttpClient();
HttpResponseMessage response = await client.PostAsync(requestUrl, null);
var taskStr = response.Content.ReadAsStringAsync();
var validationResponse = JsonSerializer.Deserialize<Dictionary<string, object>>(taskStr.Result);

return validationResponse.ContainsKey("success") && bool.TryParse(validationResponse["success"].ToString(), out var result) && result;
```

**Comment by Serk352** [comment]
Hi @jagbarcelo,

This bug was fixed in #490.
